### ISSUE TITLE
Specify max check output length in config

### ISSUE THREAD

**Issue by lewit** [sensu/sensu#1074]
Currently sensu server truncates the check output of metric checks to 255 characters when saving to redis, and this limit is hardcoded in the lib/sensu/server/process.rb module:

output_lines = check[:output].split("\n")
output = output_lines.first || check[:output]
if output_lines.size > 1 || output.length > 255
  output = output[0..255] + "\n..."
end
check.merge(:output => output)

It would be useful if there was a way to specify this limit in the config file.

**Comment by portertech** [comment]
@franceesco What is your use case for storing more than a single line and/or 255 characters of metric check output?

**Comment by portertech** [comment]
There is currently no Sensu server configuration, outside of the standard primitive definitions (checks, handlers, etc).

**Comment by franceesco** [comment]
I use some checks that define several metrics, and I only see some of them due to the truncation, e.g.: disk_usage_metric plugin. In some cases I'd like to add check results besides metrics which also makes the output too long. For example: process time of a request with source IP and path.
Maybe I can solve the latter by adding custom fields to the event because the latest version of uchiwa-web displays the last result which could contain these attributes.

**Comment by calebhailey** [comment]
@franceesco hi there. :) 

Sorry for the radio silence on this one. Is this still an issue for you? Please advise and we'll resume discussion about a potential solution. :)

**Comment by cjchand** [comment]
@calebhailey Not a showstopper by any means, but there is some value to give Operators a readout on the status of multiple items related to a check. For example, we have a check that iterates each drive to check SMART data on those drives, printing a line per drive with the Drive ID, serial number, and SMART status summary.

The Ops folks feel better seeing each of the drives listed in the output instead of a one-liner saying "All's well!" The paranoia is that the script might be falling though the bottom and falsely reporting all clear. Having each drive listed individually makes them feel better. :)

That said, I can definitely see that there needs to be some bound so people don't stuff "War and Peace" in their check output, but wanted to toss it out there to see if it helps justify the effort.

**Comment by calebhailey** [comment]
@portertech I wonder if we should reconsider this now that we have introduced the `"sensu"` configuration scope?

**Comment by hanynowsky** [comment]
Look for our case, we are using a check that gathers system facts as json :+1: curl -s ...... | jq .
 "issued": 1477400634,
     "executed": 1477400634,
     "duration": 1.292,
     "output": "{\"loadaverage\":\"0.58 0.56 0.53 1/318 3799\",\"osfamily\":\"RedHat\",\"mysqlversion\":\"NA\",\"phpversion\":\"RedHat\",\"apacheversion\":\"NA\",\"kernel\":\"Linux\",\"blockdevice_sr0_size\":1073741312,\"blockdevice_sr0_vendor\":\"NECVMWar\",\"blockdevice_sr0_model\":\"VMware IDE CDR00\",\n...",
     "status": 0
   }
 }

The output is more than 50 lines. But it gets truncated at 256 chars! That is a frustrating limitation in our case. We need now to write a handler, tell the handler to store the date in a DB like SQL in some cluster and query it  :(

**Comment by portertech** [comment]
``` json
{
  "sensu": {
    "spawn": {
      "...": "..."
    },
    "events": {
      "truncate_check_output": 
    }
  }
}
```

**Comment by portertech** [comment]
Could add a check definition attribute, i.e `"truncate_output": false`.

**Comment by cam-itv** [comment]
Hi, this would be really helpful for us too. Is there any more word on this? Thanks

**Comment by isavcic** [comment]
Would be helpful, we're looking into exporting the ```/results``` route to Prometheus and we want to have the entire output there.

**Comment by portertech** [comment]
At a minimum, we should implement check attributes, such as `"truncate_output": false` and `"truncate_output_length": 666`.

**Comment by xyntrix** [comment]
Those options sound great!  Would it make sense to also add server specific global defaults (if defined)? That would help prevent the need to go backfill every check and still remain compatible with existing behavior (no surprises).

I'm not sure if there's a precedent for naming scheduler/handler specific options, but an example Sensu  related variable might be something like.
- checks.truncate_output_default
- checks.truncate_output_length_default

**Comment by portertech** [comment]
We do have the "sensu" global configuration scope, which currently encompasses "spawn", https://github.com/sensu/sensu-settings/blob/master/lib/sensu/settings/validators/sensu.rb We could utilize this scope for "check" or "checks".

**Comment by portertech** [comment]
Closed by #1750
